bin/download-spark.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+SPARK_VERSION="2.0.0"
+HADOOP_VERSION="2.7"
+
+FWDIR="$(dirname "${BASH_SOURCE-$0}")"
+ZEPPELIN_HOME="$(cd "${FWDIR}/.."; pwd)"
+ZEPPELIN_ENV="${ZEPPELIN_HOME}/conf/zeppelin-env.sh"
+ZEPPELIN_ENV_TEMP="${ZEPPELIN_HOME}/conf/zeppelin-env.sh.template"
+
+# Downloads file from the given URL.
+# Ties 3 times with 1s delay, 20s read and 15s connection timeouts.
+# Arguments: url - source URL
+function download_with_retry() {
+  local url="$1"
+  wget --retry-connrefused --waitretry=1 --read-timeout=20 --timeout=15 -t 3 "${url}"
```

Comment:
`wget` is not installed by default on some Linux distributions, but usually `cUrl` does. 

For compatibility reasons this code needs to:
 - check if download tool of choice is installed
 - if not - report it to the user properly (and fail gracefully?)
 - output of download command must be checked (as with [any other command with side effects](https://google.github.io/styleguide/shell.xml#Checking_Return_Values)) and in case of failure - reported to user properly